File: EvenFibonaccis.java
/**
 * This program calculates the sum of all the even fibonacci numbers below a maximum value
 */
public class EvenFibonaccis {
    public static void main(String[] args) {
        int total = 0;
        int maxValue = 4_000_000;

        int secondLast = 0;
        int last = 1;
        int temp = 0;
        while (temp <= maxValue) {
            if (temp % 2 == 0) {
                total += temp;
            }

            temp = last + secondLast;
            secondLast = last;
            last = temp;
            
            // System.out.println(last);
        }

        System.out.println(total);

    }

}

File: EvenFibonaccis.py
# This program calculates the sum of all the even fibonacci numbers below a maximum value
total = 0
maxValue = 4000000

secondLast = 0
last = 1
temp = 0

while (temp <= maxValue):
    if (temp % 2 == 0):
        total += temp
    
    temp = secondLast + last
    secondLast = last
    last = temp

print(total)